### PR TITLE
926 fix z 150 false positives

### LINKED ISSUE #926: Zalando rule 150: false-positive for `502`
How to reproduce:

```
openapi: 3.0.2
info:
  title: Some API
  version: 0.0.1
  x-api-id: xxx
  x-audience: company-internal
paths:
  'path/':
    get:
      responses:
        '502': {}
```

returns a `MUST` violation. Should return only a `MAY` violation since `502` is not in the list of "well-understood" HTTP status codes.

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Fixes #926

### CODE REVIEW

File: server/src/main/java/de/zalando/zally/rule/zalando/UseStandardHttpStatusCodesRule.kt [hunk 1]
```diff
@@ -38,13 +38,13 @@ class UseStandardHttpStatusCodesRule(@Autowired rulesConfig: Config) {
      * @param context the context to validate
      * @return list of identified violations
      */
-    @Check(severity = Severity.MUST)
+    @Check(severity = Severity.SHOULD)
```
[the remark this answers is not in the document]
Comment (maxim-tschumak) [reply server/src/main/java/de/zalando/zally/rule/zalando/UseStandardHttpStatusCodesRule.kt]: In general, I agree.

Currently, we talk about two sets of status codes in the guidelines: standard (RFC) and "well-understood". As I interpret the rule, the API designers `MUST` use standard codes and `SHOULD` try using only the "well-understood" only (which is a subset of standardized status codes).

The change I propose represents the current status of rule 150, IMO.

We can propose to add `502` and `504` to the set of "well-known" codes. Let's do this in the guidelines repo and then implement the change in Zally.

[the remark this answers is not in the document]
Comment (tkrop) [reply server/src/main/java/de/zalando/zally/rule/zalando/UseStandardHttpStatusCodesRule.kt]: Sorry, I did not looked at all necessary context. You are right.

**Comment by tkrop** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**